build.py

```diff
@@ -91,9 +91,17 @@ def install_llvm_tool(name, source_location, prefix, debug, jobs=1, clean=True,
     prefix += '/llvm-' + llvm_version
     clang_path = "{prefix}/tools/clang".format(**locals())
 
-    if not os.path.isfile(prefix + '/CMakeLists.txt'): execute('Download llvm source.', 'curl https://releases.llvm.org/{llvm_version}/llvm-{llvm_version}.src.tar.xz | tar -Jxo && mv llvm-{llvm_version}.src {prefix}'.format(llvm_version=llvm_version, prefix=prefix) )
-
-    if not os.path.isdir(clang_path): execute('Download clang source.', 'curl https://releases.llvm.org/{llvm_version}/cfe-{llvm_version}.src.tar.xz | tar -Jxo && mv cfe-{llvm_version}.src {clang_path}'.format(llvm_version=llvm_version, clang_path=clang_path) )
+    if not os.path.isfile(prefix + '/CMakeLists.txt'): 
+		execute('Download llvm source.', 'curl https://releases.llvm.org/{llvm_version}/llvm-{llvm_version}.src.tar.xz --output llvm-{llvm_version}.src.tar.xz'.format(llvm_version=llvm_version) )
+		execute('Decompress llvm source.', 'unxz  llvm-{llvm_version}.src.tar.xz'.format(llvm_version=llvm_version) )
+		execute('Decompress llvm source.', 'tar -xf  llvm-{llvm_version}.src.tar'.format(llvm_version=llvm_version) )
+		execute('Move llvm source.', 'mv llvm-{llvm_version}.src {prefix}'.format(llvm_version=llvm_version, prefix=prefix) )
+
+    if not os.path.isdir(clang_path): 
+		execute('Download clang source.', 'curl https://releases.llvm.org/{llvm_version}/cfe-{llvm_version}.src.tar.xz --output cfe-{llvm_version}.src.tar.xz'.format(llvm_version=llvm_version) )
+		execute('Decompress clang source.', 'unxz  cfe-{llvm_version}.src.tar.xz '.format(llvm_version=llvm_version) )
+		execute('Decompress clang source.', 'tar -xf  cfe-{llvm_version}.src.tar'.format(llvm_version=llvm_version) )
+		execute('Download clang source.', 'mv cfe-{llvm_version}.src {clang_path}'.format(llvm_version=llvm_version, clang_path=clang_path) )
```

Comment:
Could you please elaborate why this this change is needed? Particularly Original code used pipes to eliminate need for temporary files which will speed up things a bit.

Reply:
OK.  For some reason, the xz format is not supported on some systems in the tar and there is a separate unxz utility.

Reply:
Please note this change is not present in the binderCI repo now.